ui/app/clinical/consultation/controllers/consultationController.js

```diff
@@ -314,9 +314,17 @@ angular.module('bahmni.clinical').controller('ConsultationController',
                 return getUrl(board);
             };
 
+            var preSaveEvents = function () {
+                var observationForms = $scope.consultation.observationForms;
```

Comment:
How are we maikng sure, this runs only with form2?

Reply:
in the loop, line 323, we are having a if check  with condition, where only form2 forms will be passed. So it will not script will not be run for form1